File: sdk/python/workbuoy.py
import os, json, requests

class WorkBuoyClient:
    def __init__(self, base=None, api_key=None, tenant_id=None):
        self.base = (base or os.getenv('API_BASE_URL','http://localhost:3000')).rstrip('/')
        self.h = {'x-api-key': api_key or os.getenv('API_KEY','dev-123'),
                  'x-tenant-id': tenant_id or os.getenv('TENANT_ID','demo-tenant')}

    def list_contacts(self, limit=50):
        return requests.get(f"{self.base}/api/v1/crm/contacts", params={'limit':limit}, headers=self.h).json()

    def create_contact(self, body:dict):
        return requests.post(f"{self.base}/api/v1/crm/contacts", headers={**self.h, 'Content-Type':'application/json'}, data=json.dumps(body)).json()

    def patch_contact(self, cid:str, body:dict):
        return requests.patch(f"{self.base}/api/v1/crm/contacts/{cid}", headers={**self.h, 'Content-Type':'application/json'}, data=json.dumps(body)).json()

    def list_opportunities(self, limit=50):
        return requests.get(f"{self.base}/api/v1/crm/opportunities", params={'limit':limit}, headers=self.h).json()

    def create_opportunity(self, body:dict):
        return requests.post(f"{self.base}/api/v1/crm/opportunities", headers={**self.h, 'Content-Type':'application/json'}, data=json.dumps(body)).json()

    def patch_opportunity(self, oid:str, body:dict):
        return requests.patch(f"{self.base}/api/v1/crm/opportunities/{oid}", headers={**self.h, 'Content-Type':'application/json'}, data=json.dumps(body)).json()

    def export(self, entity:str, fmt:str='json'):
        return requests.get(f"{self.base}/api/v1/crm/export", params={'entity':entity,'format':fmt}, headers=self.h).text

File: sdk/ts/workbuoy.ts
export class WorkBuoyClient {
  base: string;
  headers: Record<string,string>;
  constructor(params: { base: string, apiKey: string, tenantId: string }) {
    this.base = params.base.replace(/\/$/,''); 
    this.headers = { 'x-api-key': params.apiKey, 'x-tenant-id': params.tenantId, 'Content-Type':'application/json' };
  }
  async listContacts(limit=50){ const r=await fetch(`${this.base}/api/v1/crm/contacts?limit=${limit}`,{headers:this.headers}); return r.json(); }
  async createContact(body:any){ const r=await fetch(`${this.base}/api/v1/crm/contacts`,{method:'POST',headers:this.headers,body:JSON.stringify(body)}); return r.json(); }
  async patchContact(id:string, body:any){ const r=await fetch(`${this.base}/api/v1/crm/contacts/${id}`,{method:'PATCH',headers:this.headers,body:JSON.stringify(body)}); return r.json(); }
  async listOpportunities(limit=50){ const r=await fetch(`${this.base}/api/v1/crm/opportunities?limit=${limit}`,{headers:this.headers}); return r.json(); }
  async createOpportunity(body:any){ const r=await fetch(`${this.base}/api/v1/crm/opportunities`,{method:'POST',headers:this.headers,body:JSON.stringify(body)}); return r.json(); }
  async patchOpportunity(id:string, body:any){ const r=await fetch(`${this.base}/api/v1/crm/opportunities/${id}`,{method:'PATCH',headers:this.headers,body:JSON.stringify(body)}); return r.json(); }
  async batchImport(entity:'contacts'|'opportunities', file:Blob, dry=false){
    const form = new FormData();
    form.append('entity', entity); form.append('dry_run', String(dry)); form.append('file', file);
    const r=await fetch(`${this.base}/api/v1/crm/import`,{method:'POST',headers:{'x-api-key':this.headers['x-api-key'],'x-tenant-id':this.headers['x-tenant-id']},body:form as any}); return r.json();
  }
  async export(entity:'contacts'|'opportunities', format:'csv'|'json'='json'){
    const r=await fetch(`${this.base}/api/v1/crm/export?entity=${entity}&format=${format}`,{headers:this.headers}); return r.text();
  }
}
